packages/react-router/index.tsx

```diff
@@ -358,7 +358,7 @@ export function useHref(to: To): string {
   let { basename, navigator } = React.useContext(NavigationContext);
   let path = useResolvedPath(to);
 
-  if (basename !== "/") {
+  if (basename !== "/" && !path.pathname.startsWith(basename)) {
```

Comment:
Hmm, this seems like it could cause a bug when the initial portion of the URL pathname just happens to match the basename as well. Unlikely, but it definitely could happen.

I wish we understood why we have this bug. I'll dig a little deeper and see if I can figure it out.

Reply:
Addressed root of the issue by memo'ing the mutated path. This ensures we only ever mutate the path once, and a new path will be generated if the "to" value is changed.

Reply:
> Addressed root of the issue by adding a memo and removing mutations on "path". This ensures we don't mutate the memo'd path while still resolving it relative to the basename